src/app/devices/views/DeviceDetails/DeviceDetailsHeader/DeviceDetailsHeader.test.tsx

```diff
@@ -31,105 +29,87 @@ describe("DeviceDetailsHeader", () => {
   it("displays a spinner as the title if device has not loaded yet", () => {
     state.device.items = [];
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter>
-          <CompatRouter>
-            <DeviceDetailsHeader
-              headerContent={null}
-              setHeaderContent={jest.fn()}
-              systemId="abc123"
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <DeviceDetailsHeader
+        headerContent={null}
+        setHeaderContent={jest.fn()}
+        systemId="abc123"
+      />,
+      { store }
     );
 
-    expect(wrapper.find("SectionHeader").prop("loading")).toBe(true);
-    expect(wrapper.find("SectionHeader").prop("subtitleLoading")).not.toBe(
-      true
-    );
+    expect(screen.getByRole("heading")).toHaveTextContent("Loading...");
```

Comment:
Might be better to stick to a single way of doing the same thing (you also use `screen.getByTestId("section-header-title")` in a test below).